Microsoft.DotNet.Interactive/CurrentVariables.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+using Microsoft.DotNet.Interactive.Formatting;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public class CurrentVariables : IEnumerable<CurrentVariable>
+    {
+        private readonly Dictionary<string, CurrentVariable> _variables = new Dictionary<string, CurrentVariable>();
+
+        public CurrentVariables(IEnumerable<CurrentVariable> variables, bool detailed)
+            : this(detailed)
+        {
+            foreach (var variable in variables)
+            {
+                _variables[variable.Name] = variable;
+            }
+        }
+
+        public CurrentVariables(Dictionary<string, CurrentVariable> variables, bool detailed)
```

Comment:
This can be `IDictionary<string, CurrentVariable>`.

Reply:
Realized I don't use this overload anymore; I'll just get rid of it.

Reply:
Even better.